src/components/layout/header/mobile-menu.ts

```
@@ -30,17 +33,25 @@
     // Toggle button click
-    this.toggleButton.addEventListener("click", () => this.toggleMenu());
+    const toggleHandler = () => this.toggleMenu();
+    this.toggleButton.addEventListener("click", toggleHandler);
+    this.eventCleanups.push(() => this.toggleButton!.removeEventListener("click", toggleHandler));
 
     // Overlay click to close
-    this.overlay?.addEventListener("click", () => this.closeMenu());
+    if (this.overlay) {
+      const overlayHandler = () => this.closeMenu();
+      this.overlay.addEventListener("click", overlayHandler);
+      this.eventCleanups.push(() => this.overlay!.removeEventListener("click", overlayHandler));
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)